op-batcher/batcher/driver.go

```diff
@@ -284,11 +288,16 @@ func (l *BatchSubmitter) loop() {
 		case <-ticker.C:
 			l.loadBlocksIntoState(l.ctx)
 
-			// Empty the state after loading into it on every iteration.
 		blockLoop:
 			for {
-				// Collect the output frame
-				data, id, err := l.state.TxData(eth.L1BlockRef{})
+				l1tip, err := l.l1Tip(l.ctx)
+				if err != nil {
+					l.log.Error("Failed to query L1 tip")
+					break
+				}
```

Comment:
It may be a bit excessive to query the l1 head in every loop iteration. It should be enough to use the l1 head that's queried when the tx receipt is returned.